rpcserver.go

```diff
@@ -580,42 +650,97 @@ func checkFeeRateSanity(rpcFeeRate uint32) (*chainfee.SatPerKWeight, error) {
 	}
 }
 
-// FinalizeBatch attempts to finalize the current pending batch.
-func (r *rpcServer) FinalizeBatch(_ context.Context,
-	req *mintrpc.FinalizeBatchRequest) (*mintrpc.FinalizeBatchResponse,
-	error) {
-
-	var batchSibling *asset.TapscriptTreeNodes
+// FundBatch attempts to fund the current pending batch.
+func (r *rpcServer) FundBatch(_ context.Context,
+	req *mintrpc.FundBatchRequest) (*mintrpc.FundBatchResponse, error) {
 
 	feeRate, err := checkFeeRateSanity(req.FeeRate)
 	if err != nil {
 		return nil, err
 	}
 	feeRateOpt := fn.MaybeSome(feeRate)
 
-	batchTapscriptTree := req.GetFullTree()
-	batchTapBranch := req.GetBranch()
+	tapTreeOpt, err := taprpc.UnmarshalTapscriptSibling(
+		req.GetFullTree(), req.GetBranch(),
+	)
+	if err != nil {
+		return nil, err
+	}
 
-	switch {
-	case batchTapscriptTree != nil && batchTapBranch != nil:
-		return nil, fmt.Errorf("cannot specify both tapscript tree " +
-			"and tapscript tree branches")
+	batch, err := r.cfg.AssetMinter.FundBatch(
+		tapgarden.FundParams{
+			FeeRate:        feeRateOpt,
+			SiblingTapTree: tapTreeOpt,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fund batch: %w", err)
+	}
 
-	case batchTapscriptTree != nil:
-		batchSibling, err = marshalTapscriptFullTree(batchTapscriptTree)
-		if err != nil {
-			return nil, fmt.Errorf("invalid tapscript tree: %w",
-				err)
-		}
+	// If there was no batch to fund, return an empty response.
+	if batch == nil {
+		return &mintrpc.FundBatchResponse{}, nil
+	}
 
-	case batchTapBranch != nil:
-		batchSibling, err = marshalTapscriptBranch(batchTapBranch)
+	rpcBatch, err := marshalMintingBatch(batch, req.ShortResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return &mintrpc.FundBatchResponse{
+		Batch: rpcBatch,
+	}, nil
+}
+
+func (r *rpcServer) SealBatch(ctx context.Context,
+	req *mintrpc.SealBatchRequest) (*mintrpc.SealBatchResponse, error) {
+
```

Comment:
Shouldn't this require that group witnesses re provided? Otherwise, how can you seal a batch w/o providing the witness?

Reply:
Seal with no arguments will try to produce the 'default' witness of a signature with a tapd-owned key. That would fail if tapd can't sign for all asset group keys. At that point the user could try `SealBatch` again with the required witnesses.